server/utils/pdfRenderer.ts

```diff
@@ -0,0 +1,34 @@
+import { Request, Response, NextFunction } from 'express'
+
+/*
+ * This function accepts a Gotenberg client as its only argument.
+ * It returns a handler function to render a normal HTML view template to
+ * produce the raw HTML (including images, stylesheet etc). It then uses a
+ * callback function to pass rendered HTML view into the Gotenberg client
+ * to produce and return a PDF document.
+ */
+
+// TODO: Revisit to fully define these types for the "any" placeholders
+export default function pdfRenderer() {
+  return (req: Request, res: Response, next: NextFunction) => {
+    res.renderPDF = (
+      view: string,
+      // Define the pageData as - { url: string, licence: Licence, otherData: type? }
+      pageData: Buffer,
+      options: { filename: string } = { filename: 'document.pdf' },
+    ) => {
+      res.render(view, pageData, (error: Error | null) => {
```

Comment:
I don't think this needs to be in a callback to render?

This was only required previously as we'd use nunjucks to render the template and then pass the rendered template to gotenberg 

so you can:
* remove the res.render line
* assume we won't get an error in here (any error calling the backend api would be thrown earlier)
* remove the `view` parameter.

Think it might also be nicer if we move the filename up out of opts?
`res.renderPDF('document.pdf', pageData)` ??